tests/test_pfcwd.py

```diff
@@ -0,0 +1,221 @@
+                member_groups.append(fv[1])
+            elif fv[0] == "SAI_ACL_TABLE_GROUP_MEMBER_ATTR_ACL_TABLE_ID":
+                assert fv[1] == acl_table_id
+            elif fv[0] == "SAI_ACL_TABLE_GROUP_MEMBER_ATTR_PRIORITY":
+                assert True
+            else:
+                assert False
+
+    assert set(member_groups) == set(acl_group_ids)
+
+def verify_acl_port_binding(self, dvs, bind_ports):
+    atbl = swsscommon.Table(self.adb, "ASIC_STATE:SAI_OBJECT_TYPE_ACL_TABLE_GROUP")
+    acl_table_groups = atbl.getKeys()
+    assert len(acl_table_groups) == len(bind_ports)
+
+    atbl = swsscommon.Table(self.adb, "ASIC_STATE:SAI_OBJECT_TYPE_PORT")
+    port_groups = []
+    for p in [dvs.asicdb.portnamemap[portname] for portname in bind_ports]:
+        (status, fvs) = atbl.get(p)
+        for fv in fvs:
+            if fv[0] == "SAI_PORT_ATTR_INGRESS_ACL":
+                assert fv[1] in acl_table_groups
+                port_groups.append(fv[1])
+
+    assert len(port_groups) == len(bind_ports)
+    assert set(port_groups) == set(acl_table_groups)
+
+def Verify_AclTableCreation(self, dvs):
+    # check acl table in asic db
+    acl_table_id = get_acl_table_id(self, dvs)
+    assert acl_table_id is not None
+
+    # check acl table group in asic db
+    verify_acl_group_num(2)
+
+    # get acl table group ids and verify the id numbers
+    atbl = swsscommon.Table(self.adb, "ASIC_STATE:SAI_OBJECT_TYPE_ACL_TABLE_GROUP")
+    acl_group_ids = atbl.getKeys()
+    assert len(acl_group_ids) == 2
+
+    # check acl table group member
+    verify_acl_group_member(acl_group_ids, acl_table_id)
+
+    # check port binding
+    verify_acl_port_binding(dvs, "Ethernet0")
+
+def Verify_AclRuleInPorts(self, dvs):
+   # check acl rule table in asic db
+   acl_table_id = self.get_acl_table_id(dvs)
+
+   atbl = swsscommon.Table(self.adb, "ASIC_STATE:SAI_OBJECT_TYPE_ACL_ENTRY")
+   keys = atbl.getKeys()
+
+   acl_entry = [k for k in keys if k not in dvs.asicdb.default_acl_entries]
+   assert len(acl_entry) == 1
+
+   (status, fvs) = atbl.get(acl_entry[0])
+   assert status == True
+
+   value = dict(fvs)
+   assert value["SAI_ACL_ENTRY_ATTR_TABLE_ID"] == acl_table_id
+   assert value["SAI_ACL_ENTRY_ATTR_ACTION_PACKET_ACTION"] == "SAI_PACKET_ACTION_DROP"
+   in_ports = value["SAI_ACL_ENTRY_ATTR_FIELD_IN_PORTS"]
+   assert dvs.asicdb.portnamemap["Ethernet0"] in in_ports
+
+
+# PFCWD test class
+class TestPfcWD(object):
+    def setup_db(self, dvs):
+        self.pdb = swsscommon.DBConnector(0, dvs.redis_sock, 0)
+        self.adb = swsscommon.DBConnector(1, dvs.redis_sock, 0)
+
+    @pytest.fixture(scope="module", autouse=True)
+    def enable_queue_counters(self, dvs):
+        dvs.runcmd("counterpoll queue enable")
+
+    def test_ingress_pfcwd_handler(self, dvs):
+        self.setup_db(dvs)
+        config_db = dvs.get_config_db()
+        
+        #Enable PFC at global level
+        enable_device_pfcwd(config_db)
```

Comment:
This can probably be made a fixture since it will be used throughout all of the tests.